src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -307,12 +588,56 @@ template <typename SetT, typename CreateKeyFuncT, typename KeyToStringFuncT>
 void Details::DocumentStatisticsTrainingOnlyPolicy::fit_impl(InputType const &input, CreateKeyFuncT const &createKeyFunc, KeyToStringFuncT const &keyToStringFunc) {
     SetT                                    documents;
 
-    split_temp(
-        input,
-        [&createKeyFunc, &documents](InputTypeConstIterator begin, InputTypeConstIterator end) {
-            documents.insert(createKeyFunc(begin, end));
+    if (_analyzer == AnalyzerMethod::WORD) {
```

Comment:
>if (_analyzer == AnalyzerMethod::WORD) { [](start = 4, length = 40)

The behavior captured in these if statements will be the same for the lifetime of the class. Consider creating a `std::function` set during construction that can be called here directly.

Reply:
fixed